### PR TITLE
Add upload function using backend api

### PR DESCRIPTION
**Pull request by adamhannadev**
This commit enables the user to upload an audio file from the index route to post to the /upload route.

### CODE REVIEW

**Comment by PranavMahesh1** [comment]
How can we run the backend?

**Comment by PranavMahesh1** [comment]
How can we run the backend or test this? @adamhannadev

**Comment by timliuGH** [comment]
I'm also unsure how to test upload function. /index page is blank for me

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
 
 # dependencies
+/api/node_modules
```
Comment (flozender) [inline .gitignore:4]: Looks like this didn't work, the node_modules have been pushed into the commit, we're going to have to fix this.

**Comment by flozender** [comment]
@PranavMahesh1 @timliuGH, you can test it by pulling the branch and running the server locally. To test the route in particular, please use postman and send in a post request to ```/upload```.

File: api/app.js [hunk 2]
```diff
@@ -38,4 +78,4 @@ app.use(function(err, req, res, next) {
   res.render('error');
 });
 
-module.exports = app;
+module.exports = app;
```
Comment (flozender) [inline api/app.js:81]: Couldn't we just do an ```app.listen()``` instead of this approach? It doesn't seem to work on my end.

File: api/app.js [hunk 3]
```diff
@@ -3,32 +3,72 @@ var express = require('express');
 var path = require('path');
 var cookieParser = require('cookie-parser');
 var logger = require('morgan');
+const cors = require('cors');
```
Comment (flozender) [inline api/app.js:6]: Looks like we haven't integrated cors with express yet, maybe you missed it :)

File: api/app.js [hunk 4]
```diff
@@ -3,32 +3,72 @@ var express = require('express');
...
 
 var indexRouter = require('./routes/index');
 var usersRouter = require('./routes/users');
+var mixRouter = require('./routes/mix');
 
 var app = express();
 
 // view engine setup
 app.set('views', path.join(__dirname, 'views'));
-app.set('view engine', 'jade');
```
Comment (flozender) [inline api/app.js:14]: Please remove jade from the package.json if we're no longer using it.

**Comment by adamhannadev** [comment]
Awesome comments. Thanks for your help on this. I'll make those changes today.

File: .gitignore (hunk 1 above)
Comment (adamhannadev) [reply .gitignore:4]: I think this is done now.

File: api/app.js (hunk 4 above)
Comment (adamhannadev) [reply api/app.js:14]: Done.

File: api/app.js (hunk 3 above)
Comment (adamhannadev) [reply api/app.js:6]: Done.

File: api/app.js (hunk 2 above)
Comment (adamhannadev) [reply api/app.js]: Done.

**Comment by flozender** [comment]
LGTM

**Comment by flozender** [comment]
Closes #18

**Pull request merged**